torch/optim/lbfgs.py

```diff
@@ -106,16 +278,18 @@ def step(self, closure):
         state['func_evals'] += 1
 
         flat_grad = self._gather_flat_grad()
-        abs_grad_sum = flat_grad.abs().sum()
+        opt_cond = flat_grad.abs().max() <= tolerance_grad
```

Comment:
optimal condition is based on abs_grad_max instead of abs_grad_sum, this comes from minFunc 2012 (note that minFunc 2009 used abs_grad_sum)